frontend/src/components/settings/ManageUserDialog.vue

```diff
@@ -274,6 +288,82 @@ export default {
     getAddUserSuccessMsg() {
       return ADD_USER_SUCCESS_MSG
     },
+
+    /**
+     * Returns true if users facility access has changed, false otherwise.
+     */
+    hasUserFacilityAccessChanged(selectedFacilityIds, userFacilities) {
+      if (selectedFacilityIds?.length !== userFacilities?.length) {
+        return true;
+      } else {
+        for (let facilityId of selectedFacilityIds) {
+          if (!userFacilities.includes(facilityId)) {
+            return true;
+          }
+        }
+      }
+      return false;
+    },
+
+    /**
+     * Get updated user facility access.
+     */
+    async getUpdatedFacilityAccess(user, selectedFacilityIds) {
+      let facilitiesToAdd = []
+      let facilitiesToRemove = []
+      // Get facility objects selected by facilityIds
+      const selectedFacilities = this.getSelectedFacilitiesByIds(selectedFacilityIds)
+      // Get users current facilities
+      const userFacilities = await this.getUserFacilities(user.contactId, false)
+      if (Object.keys(user.facilities).length === 0 && user.facilities.constructor === Object) {
```

Comment:
I'm not too sure what this constructor check does.

Reply:
checks facilities is an object.